state/protocol/badger/state.go

```diff
@@ -227,6 +236,15 @@ func (state *State) bootstrapSealingSegment(segment *flow.SealingSegment, head *
 			}
 		}
 
+		// root seal contains the result ID for the sealed root block. If the sealed root block is
+		// different from the finalized root block, then it means the node dynamically bootstrapped.
+		// In that case, we should index the result of the sealed root block so that the EN is able
+		// to execute the next block.
+		err := transaction.WithTx(operation.SkipDuplicates(operation.IndexExecutionResult(rootSeal.BlockID, rootSeal.ResultID)))(tx)
+		if err != nil {
+			return fmt.Errorf("could not index root result: %w", err)
+		}
```

Comment:
I think this is duplicated in the execution state bootstrap. 

From what I can tell indexing execution results is an execution-node-specific function (no other nodes read these values, and subsequent execution results are indexed from the execution state module (execution-node specific). 

Therefore I think the cleanest approach would be to:
- do not index execution results in protocol state bootstrapping
- index required execution results in execution state bootstrapping

If this is a bunch of extra work, then don't worry about it. I mention it because it seems like (from the `SkipDuplicates`) we are doing the indexing twice, and could just remove the indexing from one place.